src/app/shared-services/staypoint/staypoint.service.ts

```diff
@@ -0,0 +1,223 @@
+import { Injectable } from '@angular/core'
+import { SqliteService } from '../../shared-services/db/sqlite.service'
+import haversine from 'haversine-distance'
+import {
+  Trajectory,
+  TrajectoryData,
+  TrajectoryType,
+} from 'src/app/model/trajectory'
+import { StayPointData, StayPoints } from 'src/app/model/staypoints'
+import { TrajectoryService } from '../trajectory/trajectory.service'
+import { take } from 'rxjs/operators'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StaypointService {
+  // for meaning of these two parameters, please see detectStayPoints() documentation
+  // if you change one or both, please also update the associated detected staypoints in staypoint.service.spec.ts
+  readonly DIST_THRESH_METERS = 100
+  readonly TIME_THRESH_MINUTES = 15
+
+  constructor(
+    private db: SqliteService,
+    private trajService: TrajectoryService
+  ) {}
+
+  /**
+   * Return staypoints for trajectoryID saved in database
+   * @param trajectoryID The identifier of the trajectory to which staypoints belong
+   * @return The staypoints
+   */
+  async getStayPoints(trajectoryID: string): Promise<StayPoints> {
```

Comment:
When being meticolous, I'd say the right camel-case for this would be `trajectoryId`